src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -123,17 +123,30 @@ abstract class BTypes {
    * has the method.
    */
   val indyLambdaImplMethods: mutable.AnyRefMap[InternalName, mutable.LinkedHashSet[asm.Handle]] = recordPerRunCache(mutable.AnyRefMap())
+
+  /**
+    * add methods
+    * @return the added methods. Note the order is undefined
+    */
   def addIndyLambdaImplMethod(hostClass: InternalName, handle: Seq[asm.Handle]): Seq[asm.Handle] = {
```

Comment:
Thanks for pointing out that the order is unspecified. I was sort of worried that this is a (small) problem, and that bytecode generation of `lambdaDeserialize` will be non-deterministic, but it looks like that part is taken care of by the `LinkedHashSet` here, and the non-deterministic order in the return value is benign given the way it is used in the inliner.